internal-dns/types/src/config.rs

```diff
@@ -142,6 +144,21 @@ pub struct DnsConfigBuilder {
     /// network
     zones: BTreeMap<Zone, Ipv6Addr>,
 
+    /// a map of DNS service zone UUIDs to the address that service answers DNS
+    /// queries on.  Each DNS server is present on the control plane network
+    /// with its HTTP interface on an IPv6 address in the map above, as well as
+    /// listening on port 53 for DNS queries on a different IP address.  These
+    /// addresses may be IPv4 or IPv6, depending on if the server is for
+    /// internal or external use, and the rack's external IP range.
```

Comment:
I was pretty confused by this.  If I'm understanding the code correctly, this map only contains entries for internal DNS zones (not external ones -- `host_zone_external_dns()` doesn't touch this).  And it always contains the underlay IP of the zone.  Doesn't that make it a subset of `self.zones`?  And shouldn't the IP always be a v6 IP?

Reply:
oh, i see, yeah. directly an artifact of how i got here: this was once `dns_addresses`, then i realized `DnsConfigBuilder` has to distinguish between internal and external, and i didn't update the comment along the way.

the internal DNS server addresses are always v6, and currently this is a subset of `self.zones`. i did want to record the IPs again here, though, because i wasn't sure we want to depend on `InternalDns::dns_address` and `InternalDns::http_address` having the same IP addresses.

would it be more accurate to have the blueprint's representation of the zone have the one IP, with ports for HTTP and DNS included? i'm still a bit fuzzy on how these relate to `gz_address` and how that's set up, maybe it's more fixed than i thought.

Reply:
> i did want to record the IPs again here, though, because i wasn't sure we want to depend on InternalDns::dns_address and InternalDns::http_address having the same IP addresses.

To my knowledge it's not currently possible for these to diverge.

> would it be more accurate to have the blueprint's representation of the zone have the one IP, with ports for HTTP and DNS included? i'm still a bit fuzzy on how these relate to gz_address and how that's set up, maybe it's more fixed than i thought.

Probably, and we probably discussed this when we hoisted the underlay IP out of the individual zone types and into the `BlueprintZoneConfig`, but I think it was more trouble than it was worth to fix.